pkg/provider/cloud/openstack/provider.go

```diff
@@ -746,31 +747,43 @@ func GetCredentialsForCluster(cloud kubermaticv1.CloudSpec, secretKeySelector pr
 		}
 	}
 
-	if tenant == "" && cloud.Openstack.CredentialsReference != nil && cloud.Openstack.CredentialsReference.Name != "" {
-		tenant, err = secretKeySelector(cloud.Openstack.CredentialsReference, resources.OpenstackTenant)
-		if err != nil {
+	if project == "" && cloud.Openstack.CredentialsReference != nil && cloud.Openstack.CredentialsReference.Name != "" {
+		if project, err = firstKeyOrFallback(secretKeySelector, cloud.Openstack.CredentialsReference, resources.OpenstackProject, resources.OpenstackTenant); err != nil {
 			return &resources.OpenstackCredentials{}, err
 		}
 	}
 
-	if tenantID == "" && cloud.Openstack.CredentialsReference != nil && cloud.Openstack.CredentialsReference.Name != "" {
-		tenantID, err = secretKeySelector(cloud.Openstack.CredentialsReference, resources.OpenstackTenantID)
-		if err != nil {
+	if projectID == "" && cloud.Openstack.CredentialsReference != nil && cloud.Openstack.CredentialsReference.Name != "" {
+		if projectID, err = firstKeyOrFallback(secretKeySelector, cloud.Openstack.CredentialsReference, resources.OpenstackProjectID, resources.OpenstackTenantID); err != nil {
 			return &resources.OpenstackCredentials{}, err
 		}
 	}
 
 	return &resources.OpenstackCredentials{
 		Username:                    username,
 		Password:                    password,
-		Tenant:                      tenant,
-		TenantID:                    tenantID,
+		Project:                     project,
+		ProjectID:                   projectID,
 		Domain:                      domain,
 		ApplicationCredentialID:     applicationCredentialID,
 		ApplicationCredentialSecret: applicationCredentialSecret,
 	}, nil
 }
 
+// firstKeyOrFallback read the secret and return value for the firstkey. if the firstKey does not exist, tries with
+// fallbackKey. if the fallbackKey does not exist then return an error
+func firstKeyOrFallback(secretKeySelector provider.SecretKeySelectorValueFunc, configVar *providerconfig.GlobalSecretKeySelector, firstKey string, fallbackKey string) (string, error) {
```

Comment:
s/firstKeyOrFallback/getFirstKey

I think we do not need to inform about fallback in the method name.

Reply:
done :)